test/test_other.py

```diff
@@ -11327,6 +11327,18 @@ def test_old_makeDynCall_syntax(self):
     err = self.run_process([EMCC, test_file('test_old_dyncall_format.c'), '--js-library', test_file('library_test_old_dyncall_format.js')], stderr=PIPE).stderr
     self.assertContained('syntax for makeDynCall has changed', err)
 
+  # Test that {{{ makeDynCall('sig', 'this.foo') }}} macro works, i.e. when 'this.' is referenced inside the macro block.
+  @parameterized({
+    'plain': [[]],
+    'dyncalls': [['-sDYNCALLS']]})
+  def test_this_in_dyncall(self, args):
+    self.do_run_in_out_file_test(test_file('no_this_in_dyncall.c'), emcc_args=['--js-library', test_file('no_this_in_dyncall.js')] + args)
+
+  @requires_v8
+  def test_this_in_dyncall_memory64(self):
+    self.v8_args += ['--experimental-wasm-memory64']
+    self.do_run_in_out_file_test(test_file('no_this_in_dyncall.c'), emcc_args=['--js-library', test_file('no_this_in_dyncall.js'), '-sMEMORY64', '-Wno-experimental'])
```

Comment:
Maybe a comment as to why we need to run this test in memory64 mode?  (its not obvious to me).

Reply:
That test is in order since enabling MEMORY64 also creates an anonymous enclosing function in makeDynCall:

https://github.com/emscripten-core/emscripten/blob/e9ba3598df8f219b8c9c505c4ed1ed4618001123/src/parseTools.js#L715

https://github.com/emscripten-core/emscripten/blob/e9ba3598df8f219b8c9c505c4ed1ed4618001123/src/parseTools.js#L779-L781